test/click.jest.js

```diff
@@ -307,6 +307,193 @@ describe('Page.click', function() {
+    });
+    await page.goto(server.EMPTY_PAGE);
+    await page.setViewportSize({width:1920, height:1080});
+    await page.setContent(`
+    <div>
+        <iframe src="${server.CROSS_PROCESS_PREFIX + '/frames/link.html'}"  width="100%" height="100%" > </iframe>
+    </div>
+    `)
+    const frame = page.frames()[1];
+    const link = await frame.$("#link-target");
+    await link.click();
+    const msg = await clickNotification;
+    expect(msg).toBe("47");
+  })
+  it('should click the visible link in an iframe (cross-origin) in fixed position div', async({page, server}) => {
+    const clickNotification = new Promise(fulfill => {
+      page.on('dialog', dialog => {
+        fulfill(dialog.message());
+        dialog.accept();
+      });
+    });
+    await page.goto(server.EMPTY_PAGE);
+    await page.setViewportSize({width:1920, height:1080});
+    await page.setContent(`
+    <div style="position:fixed;top:300px;left:300px;width:400px;height:400px;">
+        <iframe src="${server.CROSS_PROCESS_PREFIX + '/frames/link.html'}"  width="100%" height="100%" > </iframe>
+    </div>
+    `)
+    const frame = page.frames()[1];
+    const link = await frame.$("#link-target");
+    await link.click();
+    const msg = await clickNotification;
+    expect(msg).toBe("47");
+  })
+  it('should click and navigate to a x-frame-options:DENY link in fixed position div', async({page, server}) => {
+    server.setRoute('/login-with-x-frame-options-deny.html', async (req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.setHeader('X-Frame-Options', 'DENY');
+      res.end();
+    });
+
+    server.setRoute('/wikipedia.html', async(req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.end(`
+      <!DOCTYPE html>
+      <html>
+        <body>
+          <a id="pt-login" href="/login-with-x-frame-options-deny.html">login</a>
+        </body></html>
+      `)
+    })
+
+    server.setRoute('/wrapper.html', async(req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.end(`
+      <!DOCTYPE html>
+      <html>
+      <body>
+          <div style="position:fixed;top:300px;left:300px;width:400px;height:400px;">
+              <iframe src="${server.CROSS_PROCESS_PREFIX + '/wikipedia.html'}"  width="100%" height="100%" > </iframe>
+          </div>
+      </body></html>
+      `)
+    })
+
+    await page.goto(server.PREFIX + '/wrapper.html')
+    const navigated = new Promise(fulfull => {
+      page.on('framenavigated', (frame) => {
+        if (frame.url().endsWith('/login-with-x-frame-options-deny.html')) {
+          fulfull(frame.url());
+        }
+      });
+    });
+
+    const consoleMessaged = new Promise(fulfill => {
+      page.on('console', msg => {
+        fulfill(msg.text());
+      });
+    });
+
+    const frame = page.frames()[1];
+    const button = await frame.$('#pt-login');
+    await button.click();
+
+    if (FFOX) {
+      expect(await navigated).toBeTruthy();
```

Comment:
Consumer Firefox would actually block this navigation; however, Playwright-Firefox does not block it. Should Playwright-Firefox block it?

Reply:
I think this is a bug. We should mark the test as fail(FFOX).